### PR TITLE
fix: store the dev environment next to other environments in the .jx folder

### LINKED ISSUE #3741: jx install --ng fails when deploying tekton
### Summary 

A `jx install --ng` fails when deployed using a vanilla `.jx` and a vanilla GKE cluster.

### Steps to reproduce the behavior
1) stand a new GKE cluster up using terraform 
2) remove the jx dir in $HOME
`rm --recursive --force ~/.jx`
3) reinstall jx as per guide 
4) begin an install using the following comand:
`jx install --namespace=serverless --ng --environment-git-owner=[redacted] --domain=[redacted] --default-environment-prefix=releng --git-private --git-username=[redacted] --git-api-token=[redacted] --provider=gke --ingress-namespace=kube-system --ingress-deployment=nginx-ingress-controller --ingress-service=nginx-ingress-controller`

### Expected behavior
jx installs successfully on a GKE cluster

### Actual behavior
The installation fails with this error
```
error: applying the GitOps development environment config: appyting the dev environment configuration: building helm chart in dir /home/null/jenkins-x-dev-environment/env: failed to lint the chart '/home/null/jenkins-x-dev-environment/env': failed to run 'helm lint' command in directory '/home/null/jenkins-x-dev-environment/env', output: '==> Linting .
[ERROR] templates/: render error in "env/charts/tekton/templates/250-git-secret.yaml": template: env/charts/tekton/templates/250-git-secret.yaml:4:43: executing "env/charts/tekton/templates/250-git-secret.yaml" at <b64enc>: invalid value; expected string
```

### Jx version

The output of `jx version` is:

```
NAME               VERSION
jx                 2.0.38
Kubernetes cluster v1.12.7-gke.10
kubectl            v1.14.1
helm client        v2.13.1+g618447c
helm server        v2.13.1+g618447c
git                git version 2.20.1
Operating System   Fedora release 29 (Twenty Nine)
```

### Jenkins type

- [x] Next Generation (Tekton + Prow)
- [ ] Classic Jenkins
- [] Serverless Jenkins (JenkinsFileRunner + Prow)

### Kubernetes cluster
GKE created using TF

### Operating system / Environment
Fedora release 29 (Twenty Nine)

### PR DESCRIPTION
**Pull request by ccojocar**
#### Submitter checklist

- [x] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description

Store and name the dev environment folder consistently with other environments.

```
.jx
├── environments
│   └── ccojocar
│       ├── environment-cosmin-devenv12345-dev
│       ├── environment-cosmin-devenv12345-production
│       └── environment-cosmin-devenv12345-staging
```

#### Which issue this PR fixes

fixes #3741 #3695 #3694 #3684

### CODE REVIEW

File: pkg/jx/cmd/install.go [hunk 1]
```diff
@@ -1553,7 +1553,12 @@ func (options *InstallOptions) configureGitOpsMode(configStore configio.ConfigSt
 				return "", "", err
 			}
 		}
-		gitOpsDir = filepath.Join(options.Flags.Dir, "jenkins-x-dev-environment")
+
+		if options.Flags.DefaultEnvironmentPrefix == "" {
+			options.Flags.DefaultEnvironmentPrefix = strings.ToLower(randomdata.SillyName())
```
Comment (hferentschik) [inline pkg/jx/cmd/install.go]: If you search for the usage of `options.Flags.DefaultEnvironmentPrefix` you find that we assign a value to it at multiple places. I think that is confusing and the prefixes might in the end not match up (which would be nice).

Would it not be better to have a `ValidateOptions` function of the sort at the beginning of `Run` and set a valid `DefaultEnvironmentPrefix` once?

Comment (ccojocar) [reply pkg/jx/cmd/install.go]: good call! I moved it into check flags.

File: pkg/jx/cmd/install.go [hunk 2]
```diff
@@ -438,6 +438,12 @@ func (options *InstallOptions) checkFlags() error {
 		flags.ExposeControllerURLTemplate = "{{.Service}}-{{.Namespace}}.{{.Domain}}"
 	}
 
+	// Make sure that the default environment prefix is configured. Typically it is the cluster
```
Comment (hferentschik) [inline pkg/jx/cmd/install.go]: 👍

File: pkg/jx/cmd/install.go [hunk 3]
```diff
@@ -1630,58 +1638,70 @@ func (options *InstallOptions) generateGitOpsDevEnvironmentConfig(gitOpsDir stri
...
 			log.Infof("Pushed Git repository to %s\n\n", util.ColorInfo(repo.HTMLURL))
+
+			dir = filepath.Join(envDir, gitRepoOptions.Owner)
+			if _, err := os.Stat(dir); os.IsNotExist(err) {
+				if err := os.MkdirAll(dir, 0755); err != nil {
+					return "", errors.Wrapf(err, "creating directory %q", dir)
+				}
+			}
+			dir = filepath.Join(dir, repo.Name)
+			if err := os.Rename(gitOpsDir, dir); err != nil {
```
Comment (hferentschik) [inline pkg/jx/cmd/install.go]: @ccojocar in the integration tests you seem to be running into an equivalent of this https://github.com/jenkins-x/jx/issues/449

Comment (ccojocar) [reply pkg/jx/cmd/install.go]: It seems so. I am trying to use the utility method instead.

**Pull request merged**